Applied https://github.com/eclipse-ee4j/glassfish-grizzly-ahc/pull/45

carryel opened the pull request:
https://github.com/eclipse-ee4j/glassfish-grizzly-ahc/pull/45 "Java Security Ultimate Security Repo Scanner 2023"

carryel committed:
Applied https://github.com/eclipse-ee4j/glassfish-grizzly-ahc/pull/45

carryel commented:
It looks good to me because most of them are test codes, and I know that Files.createTempFile is more secure than File.createTempFile.

pzygielo commented on src/main/java/com/ning/http/client/multipart/MultipartUtils.java:
```diff
@@ -95,7 +97,7 @@ public static MultipartBody newMultipartBody(List<Part> parts, FluentCaseInsensi
 
     private static byte[] generateMultipartBoundary() {
-        Random rand = new Random();
+        Random rand = new SecureRandom();
```
Not sure if creating brand new `SecureRandom` for every single invocation of `generateMultipartBoundary` is optimal. Or even needed.

If I understand correctly, this boundary only needs to be unique within some container. Does it need to be _cryptographically secure_?

carryel replied:
Let me look into that a little more.

carryel committed:
Applied https://github.com/eclipse-ee4j/glassfish-grizzly-ahc/pull/45 "Java Security Ultimate Security Repo Scanner 2023"
+ Applied the review of https://github.com/eclipse-ee4j/glassfish-grizzly-ahc/pull/128#discussion_r2455892868.

carryel commented:
@pzygielo I hope this change is also ok, correct? If you have time later, I'd really appreciate it if you could review it!
